### PR TITLE
fix: ant build Sbom on Windows

### PR DESCRIPTION
**Pull request by zdtsw**
Fix:
- Ant does not recognize cygwin path, add logic check if it is running on Windows
  https://ant.apache.org/manual/platform.html
  running test: https://ci.adoptopenjdk.net/view/Failing%20Temurin%20jobs/job/build-scripts/job/jobs/job/jdk/job/jdk-windows-x64-temurin/82/console with debug info enabled
- uplift cyclondx-core and jackson SDK to latest version
- uplift to new schema to use FRAMEWORK than APPLICATION
- update to use eclipse than adoptium as vendor

Test on
[non-win](https://ci.adoptopenjdk.net/view/Failing%20Temurin%20jobs/job/build-scripts/job/jobs/job/jdk/job/jdk-linux-x64-temurin/168/)   => works
[win](https://ci.adoptopenjdk.net/view/Failing%20Temurin%20jobs/job/build-scripts/job/jobs/job/jdk/job/jdk-windows-x86-32-temurin/58/console) => failed in build jdk because the nightly new code from jdk
on final commit

Fixes: https://github.com/adoptium/temurin-build/issues/2927

### CODE REVIEW

File: sbin/build.sh [hunk 1]
```diff
@@ -667,10 +673,17 @@ generateSBoM() {
...
-  if [[ "${BUILD_CONFIG[OS_KERNEL_NAME]}" =~ .*cygwin.* ]]; then
-    classpath="${classpath//jar:/jar;}"
+  if [[ "$OSTYPE" == "cygwin" ]] || [[ "$OSTYPE" == "msys" ]]; then
+    for jarfile in "${CYCLONEDB_DIR}/build/jar/temurin-gen-sbom.jar" "${CYCLONEDB_DIR}/build/jar/cyclonedx-core-java.jar" "${CYCLONEDB_DIR}/build/jar/jackson-core.jar" "${CYCLONEDB_DIR}/build/jar/jackson-dataformat-xml.jar" "${CYCLONEDB_DIR}/build/jar/jackson-databind.jar" "${CYCLONEDB_DIR}/build/jar/jackson-annotations.jar" "${CYCLONEDB_DIR}/build/jar/json-schema.jar" "${CYCLONEDB_DIR}/build/jar/commons-codec.jar" "${CYCLONEDB_DIR}/build/jar/commons-io.jar";
+    do
+      classpath+=$(cygpath -m "${jarfile}")";"
+    done
   fi
 
+  echo "DEBUG START:"
```
Comment (karianna) [inline sbin/build.sh:683]: Wrap in a debug switch?

**Review by karianna — CHANGES_REQUESTED** [review]

File: sbin/build.sh (hunk 1 above)
Comment (zdtsw) [reply sbin/build.sh:683]: should have this one set to draft, and cleanup debug when it is ready

**Comment by zdtsw** [comment]
@andrew-m-leonard / @smlambert can you do a review on this too ?

**Review by andrew-m-leonard — APPROVED** [review]

**Comment by zdtsw** [comment]
Can someone help to merge it, please :)

**Pull request merged**